docs/components/index.md

```diff
@@ -0,0 +1,315 @@
+---
+title: Blazor components
+author: guardrex
+description: Learn how to create and use Blazor components, the fundamental building blocks of Blazor apps provided by compiled Razor or C# files.
+manager: wpickett
+ms.author: riande
+ms.custom: mvc
+ms.date: 04/09/2018
+ms.prod: asp.net-core
+ms.technology: aspnet
+ms.topic: article
+uid: client-side/blazor/components/index
+---
+# Blazor components
+
+By [Luke Latham](https://github.com/guardrex)
+
+[!INCLUDE[](~/includes/blazor-preview-notice.md)]
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/client-side/blazor/components/common/samples/)
+
+To download the sample:
+
+1. Download and unzip the [Blazor.Docs repository zip file](https://codeload.github.com/aspnet/Blazor.Docs/zip/master).
+1. Use the URL in the [sample link](https://github.com/aspnet/Docs/tree/master/aspnetcore/client-side/blazor/components/common/samples/) to navigate to the sample folder.
+
+A Blazor *component* represents any feature or collection of features that can be represented by a [Razor file (\*.cshtml)](https://docs.microsoft.com/aspnet/core/mvc/views/razor) or a [C# file (\*.cs)](/dotnet/csharp/getting-started/) compiled into a C# class assembly. Components are typical features found in web app client UIs, such as a page, dialog, or form, along with its processing logic. A component can also consist entirely of programming logic without UI elements. Components are the fundamental building blocks of Blazor apps and can be nested, reused, and shared between projects.
```

Comment:
The first sentence is quite complex. How about:

Blazor applications are built using *components*. A Blazor component is a self-contained chunk of user interface (UI), such as a page, dialog, or form. It includes both the HTML markup to be rendered along with processing logic needed to inject data or respond to UI events. Typically, you'll write components using Razor files (`*.cshtml`). Components are flexible and lightweight, and can be nested, reused, and shared between projects.